Source/SwiftLintFramework/Rules/SortedImportsRule.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  SortedImportsRule.swift
+//  SwiftLint
+//
+//  Created by Scott Berrevoets on 12/15/16.
+//  Copyright © 2016 Realm. All rights reserved.
+//
+
+import Foundation
+import SourceKittenFramework
+
+public struct SortedImportsRule: ConfigurationProviderRule, OptInRule {
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "sorted_imports",
+        name: "Sorted Imports",
+        description: "Imports should be sorted.",
+        nonTriggeringExamples: [
+            "import AAA\nimport BBB\nimport CCC\nimport DDD"
+        ],
+        triggeringExamples: [
+            "import AAA\nimport ZZZ\nimport ↓BBB\nimport CCC"
+        ]
+    )
+
+    public func validateFile(_ file: File) -> [StyleViolation] {
+        let pattern = "import\\s+\\w+"
```

Comment:
What do you think of this implementation instead?

```swift
let importRanges = file.matchPattern("import\\s+\\w+",
                                     withSyntaxKinds: [.keyword, .identifier])
let nsstringContents = file.contents.bridge()
let importLength = 6
let modulesAndOffsets: [(String, Int)] = importRanges.map { range in
    let rangeAfterImport = NSRange(location: range.location + importLength,
        length: range.length - importLength)
    let module = nsstringContents.substring(with: rangeAfterImport)
        .trimmingCharacters(in: .whitespacesAndNewlines)
    let offset = NSMaxRange(range) - module.bridge().length
    return (module, offset)
}
let modulePairs = zip(modulesAndOffsets, modulesAndOffsets.dropFirst())
let violatingOffsets = modulePairs.flatMap { previous, current in
    return current < previous ? current.1 : nil
}
return violatingOffsets.map {
    StyleViolation(ruleDescription: type(of: self).description,
                          severity: configuration.severity,
                          location: Location(file: file, characterOffset: $0))
}
```

It avoids the `flatMap`/`previousMatch` side-effect mutation, avoids some of the bridging operations, leverages the existing `matchPattern(_:withSyntaxKind:)` method and avoids the cutsey use of `defer`.